hayeon/baekjoon/큐/11866-요세푸스-문제0.js

```diff
@@ -0,0 +1,42 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const [n, k] = input[0].split(" ").map(Number);
+  // const queue = [];
+  const result = [];
+
+  //큐 생성
+  /*   for (let i = 1; i <= n; i++) {
+    queue.push(i);
+  } */
+  const queue = Array.from({ length: n }, (v, i) => i + 1);
+
+  while (queue.length !== 0) {
+    for (let i = 0; i < k - 1; i++) {
+      const preNum = queue.shift();
+      queue.push(preNum);
+    }
+    result.push(queue.shift());
+  }
+
+  console.log(`<${result.join(", ")}>`);
```

Comment:
```suggestion
  const [n, k] = input[0].split(" ").map(Number);
  const queue = Array.from({ length: n }, (v, i) => i + 1);
  const result = [];

  while (queue.length !== 0) {
    for (let i = 0; i < k - 1; i++) {
      const preNum = queue.shift();
      queue.push(preNum);
    }
    result.push(queue.shift());
  }

  console.log(`<${result.join(", ")}>`);
```
- 풀이잘봤습니다 큐를 이용해서 잘푸셨네요 🙇‍♂️
- 실제 solution 코드 내에서 불 필요한 주석 코드들은 제거해주시기 바랍니다 ~ 코멘트가 필요한 부분이라면 아래 풀이 주석에다 남겨주세요 ~
- 로직은 완전히 같은데 count라는 변수를 활용해서 if...else조건식으로도 같은 결과를 가져올 수 있습니다~!
```js
  const [n, k] = input[0].split(" ").map(Number);
  const queue = Array.from({ length: n }, (_, i) => i + 1);
  const result = [];
  let count = 0;

  while (queue.length) {
    count++;

    if (count % k === 0) {
      result.push(queue.shift());
    } else {
      queue.push(queue.shift());
    }
  }

  console.log(`<${result.join(", ")}>`);
```

Reply:
오오 count 변수 활용하는 방법도 간단하네요!!!!! 이걸로도 풀어봐야겠습니다 감사합니다!!!!우왓 짱👍🏻